### PR TITLE
Adds Sscommunications subsystem

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Communication consoles now share cooldowns on announcements.
/:cl:

Sharing cooldowns is good, having a central place to make announcements
is good. This moves some of the machinery for announcements out of the
consoles, which coincides with some plans I have for the communications
machinery of the station anyway.

Also, AIs make vox announcements. Need to clarify that.

### CODE REVIEW

**Comment by Haevacht** [comment]
What's all the new defines about.

**Comment by coiax** [comment]
FUCK MY LIFE STUPID DEFINES

**Comment by MrStonedOne** [comment]
I should point that you could get the same result with a static var on consoles =P

**Comment by coiax** [comment]
Yeah, I'm going to be doing some expansion work into SScommunications when I do my giant  communication disk stuff.

File: code/controllers/subsystem/communications.dm [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	else if(!is_silicon && nonsilicon_message_cooldown > world.time)
+		. = FALSE
+	else
+		. = TRUE
+
+/datum/subsystem/communications/proc/make_announcement(mob/living/user, is_silicon, input)
+	if(!can_announce(user, is_silicon))
+		return FALSE
+	if(is_silicon)
+		minor_announce(input,"[user.name] Announces:")
```
Comment (AnturK) [inline code/controllers/subsystem/communications.dm]: Why one of these assumes sanitized input while the other doesn't

Comment (coiax) [inline code/controllers/subsystem/communications.dm]: I don't know, I just copied the code.

**Pull request merged**